test/modules/ModuleLib/ModuleUtil.cs

```diff
@@ -69,6 +70,19 @@ public static async Task ReportTestResultAsync(TestResultReportingClient apiClie
             }
         }
 
+        public static async Task ReportTestResultAsync(TestResultReportingClient apiClient, ILogger logger, TestResultBase testResult, CancellationToken cancellationToken)
```

Comment:
why do you need to a new method just for providing cancallation token? if so, you can just do this below.
```suggestion
        public static async Task ReportTestResultAsync(TestResultReportingClient apiClient, ILogger logger, TestResultBase testResult, CancellationToken cancellationToken = default)
```